bin/gaia-marionette

```diff
@@ -108,6 +108,12 @@ fi
 # find xpcshell and put it in the path
 XPCSHELL_DIR=$(dirname "$XPCSHELLSDK")
 
+MOCHA_GREP_OPTS=""
+if [ -z "$GAIA_DEVICE_TYPE" ] || [ "$GAIA_DEVICE_TYPE" = "phone" ]; then
+  MOCHA_GREP_OPTS="--invert --grep \[tablet\]|\[tv\]"
+else
+  MOCHA_GREP_OPTS="--grep \[$GAIA_DEVICE_TYPE\]"
+fi
 
```

Comment:
Logic of this snippet is weird here.
If there is a `none` type means a general test type, we should trigger it every time regardless of which device type is specified by user.
line 115 only trigger device specified tests. It should include `none` type for general test cases.